### ISSUE TITLE
max-len: count trailing comment towards limit even if ignore* is set

### ISSUE THREAD

**Issue by dwelle** [eslint/eslint#10772]
related https://github.com/eslint/eslint/issues/9059

**What rule do you want to change?**

[`max-len`](https://eslint.org/docs/rules/max-len)

I propose that trailing comments count towards `max-len`'s `code` limit EVEN if the line contains `string`/`templateLiteral`/`RegExpLiteral`/`url`/`pattern` and respective `ignore*` flag is set to `true`.

That is, if the `max-len` violation is caused by the trailing comment, and moving the comment on separate line would solve that violation, the `ignore*` flag should not apply to that comment.

Note: the behavior with `ignorePattern` is gonna be tricky because that pattern could be matched evne in the comment itself. If this would prove hard to implement, we could not include `ignorePattern` for this change.

**Does this change cause the rule to produce more or fewer warnings?**

more

**How will the change be implemented? (New option, new default behavior, etc.)?**

Most likely new option, but could as well be default behavior (breaking change).

**Please provide some example code that this change will affect:**

Even though the following code line (50-char long) is over `30` limit, it doesn't fail because `ignoreStrings` is set.

```js
/* eslint max-len: ["error", {
    code: 30,
    ignoreStrings: true
}]*/


const a = "4209019249000"; // lasdaksldkasldkasla
```

**What does the rule currently do for this code?**

[demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG1heC1sZW46IFtcImVycm9yXCIsIHtcclxuICAgIGNvZGU6IDQwLFxyXG4gICAgaWdub3JlU3RyaW5nczogdHJ1ZSxcclxuICAgIGlnbm9yZVJlZ0V4cExpdGVyYWxzOiB0cnVlLFxyXG4gICAgaWdub3JlVGVtcGxhdGVMaXRlcmFsczogdHJ1ZSxcclxuICAgIGlnbm9yZVVybHM6IHRydWUsXHJcbiAgICBpZ25vcmVQYXR0ZXJuOiBcImFyZ3VtZW50c1wiXHJcbn1dKi9cclxuXHJcbihmdW5jdGlvbiAoKSB7XHJcbiAgICByZXR1cm4gKFxyXG4gICAgICAgIC8vIOKGkyBwYXNzZXNcclxuICAgICAgICBcImxhc2tkbGFza2RcIiwgLy8gbGFrc2xrbGFsc2tkbGFza2RsXHJcbiAgICAgICAgLy8g4oaTIHBhc3Nlc1xyXG4gICAgICAgIC9hemFkbGFza2xhLywgLy8gbGFrc2xrbGFsc2tkbGFza2RsXHJcbiAgICAgICAgLy8g4oaTIHBhc3Nlc1xyXG4gICAgICAgIGBhemFkbGFza2xhYCwgLy8gbGFrc2xrbGFsc2tkbGFza2RsXHJcbiAgICAgICAgLy8g4oaTIHBhc3Nlc1xyXG4gICAgICAgIFwiaHR0cDovL2EuYlwiLCAvLyBsYWtzbGtsYWxza2RsYXNrZGxcclxuICAgICAgICAvLyDihpMgcGFzc2VzXHJcbiAgICAgICAgYXJndW1lbnRzLCAgICAvLyBsYXNrZGxhc2tsZGFza2RsYXNsXHJcbiAgICAgICAgLy8g4oaTIGRvZXNuJ3QgcGFzc1xyXG4gICAgICAgIDQyMDkwMTkyNDkwMDAgLy8gbGFzZGFrc2xka2FzbGRrYXNsXHJcbiAgICApO1xyXG59KSgpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6MiwiZm9yLWRpcmVjdGlvbiI6MiwiZ2V0dGVyLXJldHVybiI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNsYXNzLWFzc2lnbiI6Miwibm8tY29tcGFyZS1uZWctemVybyI6Miwibm8tY29uZC1hc3NpZ24iOjIsIm5vLWNvbnNvbGUiOjIsIm5vLWNvbnN0LWFzc2lnbiI6Miwibm8tY29uc3RhbnQtY29uZGl0aW9uIjoyLCJuby1jb250cm9sLXJlZ2V4IjoyLCJuby1kZWJ1Z2dlciI6Miwibm8tZGVsZXRlLXZhciI6Miwibm8tZHVwZS1hcmdzIjoyLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZW1wdHkiOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1zZWxmLWFzc2lnbiI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjJ9LCJlbnYiOnt9fX0=)

Note: sorry for the strange code; I always write it so that it can be copied verbatim to demo/editor, and tested.

```js
/* eslint max-len: ["error", {
    code: 40,
    ignoreStrings: true,
    ignoreRegExpLiterals: true,
    ignoreTemplateLiterals: true,
    ignoreUrls: true,
    ignorePattern: "arguments"
}]*/

(function () {
    return (
        // ↓ passes
        "laskdlaskd", // lakslklalskdlaskdl
        // ↓ passes
        /azadlaskla/, // lakslklalskdlaskdl
        // ↓ passes
        `azadlaskla`, // lakslklalskdlaskdl
        // ↓ passes
        "http://a.b", // lakslklalskdlaskdl
        // ↓ passes
        arguments,    // laskdlaskldaskdlasl
        // ↓ doesn't pass
        4209019249000 // lasdaksldkasldkasl
    );
})();
```

**What will the rule do after it's changed?**

```js
/* eslint max-len: ["error", {
    code: 40,
    ignoreStrings: true,
    ignoreRegExpLiterals: true,
    ignoreTemplateLiterals: true,
    ignoreUrls: true,
    ignorePattern: "arguments"
}]*/

(function () {
    return (
        // ↓ won't pass
        "laskdlaskd", // lakslklalskdlaskdl
        // ↓ won't pass
        /azadlaskla/, // lakslklalskdlaskdl
        // ↓ won't pass
        `azadlaskla`, // lakslklalskdlaskdl
        // ↓ won't pass
        "http://a.b", // lakslklalskdlaskdl
        // ↓ won't pass
        arguments,    // laskdlaskldaskdlasl
        // ↓ doesn't pass (same behavior)
        4209019249000 // lasdaksldkasldkasl
    );
})();
```
